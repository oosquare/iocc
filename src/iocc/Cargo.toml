[package]
name = "iocc"
version.workspace = true
edition.workspace = true
license.workspace = true
authors.workspace = true
repository.workspace = true
description = "Minimal dependency injection framework"

[dependencies]
oneshot.workspace = true
parking_lot.workspace = true
snafu.workspace = true
iocc-derive = { path = "../iocc-derive" }

[dev-dependencies]
mockall.workspace = true
trybuild.workspace = true

