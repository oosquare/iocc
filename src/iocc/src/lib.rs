#![allow(clippy::new_without_default)]

pub mod container;
pub mod key;
pub mod module;
pub mod provider;
pub mod scope;
mod util;

pub use iocc_derive::component;

pub mod prelude {
    pub use crate::component;
    pub use crate::container::injector::{InjectorError, TypedInjector};
    pub use crate::container::registry::{Configurer, Registry, RegistryError};
    pub use crate::container::Container;
    pub use crate::key;
    pub use crate::module::{bind, Configuration, Module};
}
