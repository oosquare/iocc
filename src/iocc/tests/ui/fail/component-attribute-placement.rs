use iocc::prelude::*;

#[component]
struct Test;

fn main() {}
