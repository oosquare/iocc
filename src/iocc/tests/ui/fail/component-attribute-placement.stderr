error: `#[component]` should be annotated on the `impl` block
 --> tests/ui/fail/component-attribute-placement.rs:4:1
  |
4 | struct Test;
  | ^^^^^^
